From 564fc9c5123848769d21c3bbc660451e7617e4c4 Mon Sep 17 00:00:00 2001
From: Wojtek Porczyk <woju@invisiblethingslab.com>
Date: Thu, 16 Jun 2016 18:53:18 +0200
Subject: [PATCH] Remove branding code
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This code caused transient build failures when there is more than one
package that provides /etc/system-release _available_ in any of the
repositories, irrespective of _any_ dependecies whatsoever. The failure
is non-deterministic, because depends on the order of packages returned
by search query.

Automatic branding install is stupid anyway and should instead be done
by package groups, kickstarts and/or whatever. But it does not surprise
me at all, because this is package is maintained as part of Fedora/RH.

Signed-off-by: Wojtek Porczyk <woju@invisiblethingslab.com>
Signed-off-by: Frédéric Pierret (fepitre) <frederic.epitre@orange.fr>
---
 src/pylorax/treebuilder.py | 25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/src/pylorax/treebuilder.py b/src/pylorax/treebuilder.py
index 0756804c..dc8eec05 100644
--- a/src/pylorax/treebuilder.py
+++ b/src/pylorax/treebuilder.py
@@ -89,33 +89,8 @@ class RuntimeBuilder(object):
         self._runner.defaults = self.vars
         self.dbo.reset()
 
-    def _install_branding(self):
-        release = None
-        q = self.dbo.sack.query()
-        a = q.available()
-        for pkg in a.filter(provides='system-release'):
-            logger.debug("Found release package %s", pkg)
-            if pkg.name.startswith('generic'):
-                continue
-            else:
-                release = pkg.name
-                break
-
-        if not release:
-            logger.error('could not get the release')
-            return
-
-        # release
-        logger.info('got release: %s', release)
-        self._runner.installpkg(release)
-
-        # logos
-        release, _suffix = release.split('-', 1)
-        self._runner.installpkg('%s-logos' % release)
-
     def install(self):
         '''Install packages and do initial setup with runtime-install.tmpl'''
-        self._install_branding()
         if len(self._installpkgs) > 0:
             self._runner.installpkg(*self._installpkgs)
         if len(self._excludepkgs) > 0:
-- 
2.19.2

